Foundation/NSLock.swift

```diff
@@ -258,3 +302,59 @@ open class NSCondition: NSObject, NSLocking {
     
     open var name: String?
 }
+
+private func timeSpecFrom(date: Date) -> timespec? {
+    let interval = date.timeIntervalSince1970
+    guard interval >= 0 else {
+        return nil
+    }
+
+    var ts = timespec()
+    ts.tv_sec = Int(interval.rounded(.down))
+    ts.tv_nsec = Int((interval - Double(ts.tv_sec)) * 1_000_000_000.0)
+    return ts
+}
+
+#if os(OSX) || os(iOS)
+
+private func allocateTimedLockData() -> UnsafeMutablePointer<_timed_lock_data>? {
```

Comment:
in Swift, failure to allocate memory usually results in a crash. Why does this return an Optional instead? What's the caller supposed to do if this fails?

Reply:
This was copied from the `-[NSLock lock(before:)]` function but you are correct, it should be stricter with the checking, I will update it to call `fatalError()` if anything fails.